### PR TITLE
feat: migrate database after federation is enabled - FS-492

### PR DESCRIPTION
**Pull request by David-Henner**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/FS-492" title="FS-492" target="_blank">Sub-taskFS-492</a>  [iOS] Migrate databases to become federation aware
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

When we detect that federation has been enabled, we need to migrate the users and conversations of all the accounts to include the domain of the federated backend.

### Solutions

Provide an interface to migrate users and conversations to the current domain.

#### Test Coverage (Optional)

- [x] Added tests for migration logic

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [x] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [x] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Review by johnxnguyen — APPROVED** [review]

File: Source/ManagedObjectContext/NSManagedObjectContext+Federation.swift [hunk 1]
```diff
@@ -0,0 +1,51 @@
+//
+// Wire
+// Copyright (C) 2020 Wire Swiss GmbH
```
Comment (KaterinaWire) [inline Source/ManagedObjectContext/NSManagedObjectContext+Federation.swift:3]:
```suggestion
// Copyright (C) 2022 Wire Swiss GmbH
```

**Review by KaterinaWire — APPROVED** [review]

**Comment by David-Henner** [comment]
release minor

**Comment by zenkins** [comment]
Hello, @David-Henner! Merging PR for command **release minor** for PR #**1367**. Will create a **minor** release after merged.
Next version is 289.1.0.

**Comment by David-Henner** [comment]
release minor

**Comment by zenkins** [comment]
Hello, @David-Henner! Merging PR for command **release minor** for PR #**1367**. Will create a **minor** release after merged.
Next version is 289.1.0.

**Pull request merged**